### PR TITLE
Use test binary for canister sig checking

### PR DESCRIPTION
**Pull request by frederikrothenberger**
The test-state-machine-binary has recently been extended to offer canister signature checks. This allows us to remove all remaining dependencies on the IC repository.

In particular, this removes the duplicate package warnings from our build and drops a lot of dependencies (-> faster builds).

### CODE REVIEW

File: src/canister_tests/src/framework.rs [hunk 1]
```diff
@@ -513,23 +508,38 @@ pub fn time(env: &StateMachine) -> u64 {
...
+        (
+            "pubkey".to_string(),
+            Value::Bytes(signed_delegation.delegation.pubkey.clone().into_vec()),
+        ),
+        (
+            "expiration".to_string(),
+            Value::Number(signed_delegation.delegation.expiration),
+        ),
+    ];
+    let mut msg: Vec<u8> = Vec::from([(DOMAIN_SEPARATOR.len() as u8)]);
```
Comment (nmattia) [inline src/canister_tests/src/framework.rs:530]: could you add a link to the message format definition?

File: src/canister_tests/src/framework.rs [hunk 2]
```diff
@@ -513,23 +508,38 @@ pub fn time(env: &StateMachine) -> u64 {
...
+pub fn verify_delegation(
+    env: &StateMachine,
+    user_key: UserKey,
+    signed_delegation: &SignedDelegation,
+    root_key: &[u8],
+) {
+    const DOMAIN_SEPARATOR: &[u8] = b"ic-request-auth-delegation";
+
+    // calculate the delegation hash (which is the signed message)
+    let map = vec![
```
Comment (nmattia) [inline src/canister_tests/src/framework.rs:520]: nit: `map` is not very description and is almost exclusively used for `map: (A -> B) -> [A] -> [B]`

**Review by nmattia — APPROVED** [review]
nice, thanks!

File: src/canister_tests/src/framework.rs [hunk 3]
```diff
@@ -513,23 +508,40 @@ pub fn time(env: &StateMachine) -> u64 {
...
-        public_key_bytes_from_der(user_key.as_ref()).unwrap(),
-        &parse_threshold_sig_key_from_der(root_key).unwrap(),
+    env.verify_canister_signature(
+        msg,
+        signed_delegation.signature.clone().into_vec(),
+        user_key.into_vec(),
+        root_key.to_vec(),
     )
-    .expect("signature invalid");
+    .expect("canister signature invalid");
```
Comment (przydatek) [inline src/canister_tests/src/framework.rs:544]: given the name of the function, maybe use "delegation signature invalid" instead?

**Review by przydatek — APPROVED** [review]
LGTM, with minor suggestions.
Thanks!

**Pull request merged**